test/intl402/Locale/constructor-options-variants-valid.js

```diff
@@ -0,0 +1,50 @@
+// Copyright 2025 Richard Gibson. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-intl.locale
+description: >
+    Checks error cases for the options argument to the Locale
+    constructor.
+info: |
+    Intl.Locale( tag [, options] )
+    12. Set _tag_ to ? UpdateLanguageId(_tag_, _options_).
+
+    UpdateLanguageId ( tag, options )
+    8. Let _variants_ be ? GetOption(_options_, *"variants"*, ~string~, ~empty~, GetLocaleVariants(_baseName_)).
+    ...
+    13. If _variants_ is not *undefined*, set _newTag_ to the string-concatenation of _newTag_, *"-"*, and _variants_.
+
+features: [Intl.Locale]
+---*/
+
+const validVariantsOptions = [
+  ['en', undefined, undefined],
+  ['en', 'spanglis', 'en-spanglis'],
+];
+for (const [lang, variants, baseName] of validVariantsOptions) {
+  let options = { variants };
+  let optionsRepr = `{variants: ${variants == null ? variants : `"${variants}"`}}`;
```

Comment:
Nitpick: the `==` operator here is very subtle but actually significant and needed. I overlooked it in several reads, and thought the code looked wrong. Would there be a more obvious way to write it?